Source/ACE/Entity/Player.cs

```diff
@@ -1605,5 +1605,30 @@ public void ClearDelayedTeleport()
             DelayedTeleportTime = DateTime.MinValue;
             WaitingForDelayedTeleport = false;
         }
+
+        override public void GameLoopUpdate()
+        {
+            uint oldHealth = Health.Current;
```

Comment:
Most discussion happens in discord - you will certainly get the most feedback and active discussion there.

To the point of the PR: I had envisioned the "IsDirty" thing because you can just set-and-forget.  It doesn't require locking.  If it sets erroneously, no big deal, and extra value to the client.